plan/logical_plan_builder.go

```diff
@@ -660,15 +660,10 @@ func (b *planBuilder) buildProjection4Union(u *LogicalUnionAll) {
 		if _, isProj := child.(*LogicalProjection); needProjection || !isProj {
 			b.optFlag |= flagEliminateProjection
 			proj := LogicalProjection{Exprs: exprs}.init(b.ctx)
-			if childID == 0 {
-				for _, col := range unionSchema.Columns {
-					col.UniqueID = b.ctx.GetSessionVars().AllocPlanColumnID()
```

Comment:
what if all children of `LogicalUnionAll` have same RetType for each Column? We would still generate new `UniqueID` for each Column? Can we just use previous `UniqueID` in this case?

Reply:
We can